src/AbstractDbEntityFetcher.php

```diff
@@ -75,19 +75,19 @@ protected function getLimitSql($limit, array $pageItem = [])
     /**
      * Helper method to get pagination result array if pagination is requested.
      *
-     * @param array $objects
+     * @param array $rows
      * @param bool  $pagination
      * @return array
      */
-    protected function getFetchPaginationResult(array $objects, $pagination)
+    protected function getFetchPaginationResult(array $rows, $pagination)
     {
         if ($pagination) {
             $totalRowCount = $this->db->fetchValue('SELECT FOUND_ROWS()');
 
-            return [$objects, $totalRowCount];
-        } else {
-            return $objects;
+            return [$this->getDbEntitiesFromRows($rows), $totalRowCount];
```

Comment:
Remember, this is public repository so please keep discussions here to English and avoid referring to private issues.

I think it could be a nice change but it is also a backward compatibility break, which means that all code using this method have to be changed. Is it worth to change all existing uses for a new edge case? There is, as @felixsand mentions, also some value to have these methods separated.

If you decide to continue with this, the method name needs to be changed to better reflect the new behavior.